Data/Code license clarification

mnalis opened the pull request:
- data (`*.txt` files) is licensed under CC0-1.0
- code is licensed under Apache-2.0

(uses https://reuse.software / SPDX)

fixes https://github.com/mnalis/StreetComplete-taginfo-categorize/issues/33

mnalis committed:
Add licenses

mnalis committed:
specify licensing - work in progress

mnalis committed:
rename license files
as per "reuse lint" recommendation

mnalis committed:
license perl code permissively (Apache-2.0), only author

mnalis committed:
update SPDX on makefiles and perl scripts

mnalis committed:
SPDX, copyright clarifications

mnalis committed:
use unmodified license text
from "reuse download"

mnalis committed:
add contributor rugk
fix to docs were subsequently replaced, so no copyright remains. Still, acknowledge for past contribution!

mnalis committed:
indicate mostly done

mnalis committed:
automatically generated files - CC0 license
see https://reuse.software/faq/#uncopyrightable

mnalis committed:
partial annotation of keys.txt copyrights
see TODO.md for missing data

mnalis committed:
REUSE.toml TODO

mnalis committed:
use REUSE.toml to consolidate same license info for *.txt files
(instead of separate *.txt.license files which need to be maintained in sync)

mnalis committed:
clarify comment

FloEdelmann commented on README.md:
```diff
@@ -1,3 +1,12 @@
+<!--
+SPDX-FileCopyrightText: 2021,2022,2024,2025 Matija Nalis <mnalis-git@voyager.hr>
+SPDX-FileCopyrightText: 2022 Flo Edelmann <florian-edelmann@online.de>
```
```suggestion
SPDX-FileCopyrightText: 2022 Flo Edelmann <git@flo-edelmann.de>
```

mnalis committed:
Update README.md

FloEdelmann preferred email

Co-authored-by: Flo Edelmann <florian-edelmann@online.de>

mnalis committed:
licensing confirmed by rugk
https://github.com/mnalis/StreetComplete-taginfo-categorize/issues/33#issuecomment-3532263573

mnalis committed:
license approval by final remaining contributors waterced & smootheFiets
https://github.com/mnalis/StreetComplete-taginfo-categorize/issues/33#issuecomment-3531164762
https://github.com/mnalis/StreetComplete-taginfo-categorize/issues/33#issuecomment-3532263573

mnalis committed:
sort by starting year
